libtock/services/alarm.c

```diff
@@ -7,6 +7,35 @@ static int within_range(uint32_t a, uint32_t b, uint32_t c) {
   return (b - a) < (b - c);
 }
 
+/** \brief Convert milliseconds to clock ticks
+ *
+ * WARNING: This function will assert if the output
+ * number of ticks overflows `UINT32_MAX`.
+ *
+ * This conversion is accurate to within 1 millisecond of a true
+ * fractional conversion.
+ *
+ * \param ms the milliseconds to convert to ticks
+ * \return ticks a number of clock ticks that
+ * correspond to the given number of milliseconds
+ */
+static uint32_t ms_to_ticks(uint32_t ms) {
+  // This conversion has a max error of 1ms.
+  // View the justification here https://github.com/tock/libtock-c/pull/434
+  uint32_t frequency;
+  libtock_alarm_command_get_frequency(&frequency);
+
+  uint32_t seconds                 = ms / 1000;
+  uint32_t leftover_millis         = ms % 1000;
+  uint32_t milliseconds_per_second = 1000;
+
+  uint64_t ticks = (uint64_t) seconds * frequency;
+  ticks += ((uint64_t) leftover_millis * frequency) / milliseconds_per_second;
```

Comment:
I'm just catching up to speed, but given that we're in 64-bit arithmetic now, why is this still dealing with seconds and fractional seconds individually? Wouldn't this suffice now?

```
    ms = (ticks * milliseconds_per_second) / freq
    ticks = (ms * freq) / milliseconds_per_second
```